### DISCUSSION TITLE
How to add a filter to an attribute of BelongsTo relationship model

### DISCUSSION THREAD

**Discussion by MagicDidou** [filamentphp/filament#13460] [Help] [answered]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

Hi everybody,

Here is the thing:
* I have a `Operator` model which has a `active` attribute which may be `true` or `false`
* I have a `Salary` model which has a relation `BelongsTo` an `Operator` (as depicted in the following code)

```php
class Salary extends Model
{
    use HasFactory;

    public function operator():BelongsTo
    {
        return $this->belongsTo(Operator::class) ;
    }
...
```

In the `SalaryResource` table view, I display the operator name and some attributes of the `Salary` model. I would like to add a filter on this table to display only `Salary` rows of operators which are `active = true`

I'm sure this is simple but I'm turning around since couple days...

Any help will be appreciated.
Thanks

**Comment by leandrocfe** [comment]
check this section
https://filamentphp.com/docs/3.x/tables/filters/select#customizing-the-select-filter-relationship-query

**Reply by MagicDidou** [reply]
Hi,

I don't think this may match what I'm trying to do. I do not want to populate the select component with a query result. What I have done so far is this:
```php
   ...
    ->filters([
        Tables\Filters\SelectFilter::make('operator.active')
            ->label('Status of the operator')
            ->options(['Not active', 'Active'])
    ])
   ...
```
but I don't know how to update the table content if 'Not active' is selected (which is 0) and if 'Active' is selected (which is 1).

**Comment by MagicDidou** [comment, marked as answer]
Hi,

I finally succeeded in doing what I wanted by using a `TernaryFilter` and customizing the queries. Here is the code of the filter :
```php
  ->filters([
      Tables\Filters\TernaryFilter::make('operator_id')
          ->label('Opérateurs actifs')
          ->queries(
              true: fn(Builder $query) => $query->whereIn('operator_id',Operator::select('id')->where('active','=','true')->get()),
              false: fn(Builder $query) => $query->whereIn('operator_id',Operator::select('id')->where('active','=','false')->get()),
              blank:fn(Builder $query) => $query,
          )
  ])
```

This may help someone else !

Thanks
